match/tensordata.py

```diff
@@ -132,15 +133,18 @@ def item(self):
             "only one element tensors can be converted into python scalars"
         )
 
-    def __reshape(self, shape: tuple):
+    def reshape_(self, shape: tuple):
```

Comment:
We'll eventually want to organize all modifying functions together (probably). You can decide when is best to do so (or if we should).